Teacher/Teacher/Submissions/SubmissionsList/View/SubmissionsFilterView.swift

```diff
@@ -0,0 +1,130 @@
+    case needsGrading
+    case notSubmitted
+    case graded
+
+    var title: String {
+        switch self {
+        case .all:
+            String(localized: "All Submissions", bundle: .teacher)
+        case .needsGrading:
+            String(localized: "Needs Grading", bundle: .teacher)
+        case .notSubmitted:
+            String(localized: "Not Submitted", bundle: .teacher)
+        case .graded:
+            String(localized: "Graded", bundle: .teacher)
+        }
+    }
+
+    var filters: [GetSubmissions.Filter] {
+        switch self {
+        case .all:
+            return []
+        case .needsGrading:
+            return [.needsGrading]
+        case .notSubmitted:
+            return [.notSubmitted]
+        case .graded:
+            return [.graded]
+        }
+    }
+}
+
+struct SubmissionsFilterView: View {
+
+    @Environment(\.viewController) private var controller
+
+    @ObservedObject private var viewModel: SubmissionListViewModel
+    private let filterOptions: SingleSelectionOptions
+
+    init(viewModel: SubmissionListViewModel) {
+        self.viewModel = viewModel
+
+        let color = viewModel.course.flatMap({ Color(uiColor: $0.color) })
+        let initialMode = viewModel.filterMode
+
+        self.filterOptions = SingleSelectionOptions(
+            all: SubmissionFilterMode.allCases.map {
+                OptionItem(id: $0.rawValue, title: $0.title, color: color)
+            },
+            initial: OptionItem(id: initialMode.rawValue, title: initialMode.title)
+        )
+    }
+
+    var body: some View {
+        VStack {
+            SingleSelectionView(
+                title: String(localized: "Submission Filter", bundle: .teacher),
+                accessibilityIdentifier: "SubmissionsFilter.filterOptions",
+                options: filterOptions
+            )
+            Spacer()
+        }
+        .background(Color.backgroundLightest)
+        .toolbar {
+            ToolbarItem(placement: .topBarTrailing) {
+                Button(
+                    action: {
+                        viewModel.filterMode = selectedFilterMode
+                        controller.value.dismiss(animated: true)
+                    },
+                    label: {
+                        Text("Done", bundle: .teacher)
+                            .font(.semibold16)
+                            .foregroundColor(color)
+                    }
+                )
+            }
+
+            ToolbarItem(placement: .topBarLeading) {
+                Button(
+                    action: { controller.value.dismiss(animated: true) },
+                    label: {
+                        Text("Cancel", bundle: .teacher)
+                            .font(.regular16)
+                            .foregroundColor(color)
+                    }
+                )
+            }
+        }
+        .navigationBarTitleView(
+            title: String(localized: "Submission list Preferences", bundle: .teacher),
```

Comment:
```suggestion
            title: String(localized: "Submission List Preferences", bundle: .teacher),
```
For consistency with filter screens in Grades, Assignments, etc
(Figma should be updated too)